reactive/kubernetes_master.py

```diff
@@ -588,6 +588,18 @@ def enable_metric_changed():
         configure_cdk_addons()
 
 
+@when("etcd.available")
+@when("config.changed.enable-csi-snapshots")
+def enable_csi_snapshots_changed():
+    """
+    Trigger CDK addon configuration if enable-csi-snapshots is updated.
+
+    :return: None
+    """
+    if is_state("leadership.is_leader"):
+        configure_cdk_addons()
```

Comment:
configure_cdk_addons runs on every hook anyway. I do not think this enable_csi_snapshots_changed handler would be needed at all.